agent/ecscni/plugin.go

```diff
@@ -111,18 +114,34 @@ func (client *cniClient) setupNS(cfg *Config) (*current.Result, error) {
 	}
 
 	seelog.Debugf("[ECSCNI] Starting ENI (%s) setup in the the container namespace: %s", cfg.ENIID, cfg.ContainerID)
-	os.Setenv("ECS_CNI_LOGLEVEL", logger.GetLevel())
-	defer os.Unsetenv("ECS_CNI_LOGLEVEL")
 
-	// Invoke eni plugin ADD command
-	result, err := client.add(runtimeConfig, cfg, client.createENINetworkConfig)
-	if err != nil {
-		return nil, errors.Wrap(err, "cni setup: invoke eni plugin failed")
+	// Invoke eni plugin ADD command based on the type of eni plugin
+	if cfg.ENIType == apieni.BranchENIType {
```

Comment:
how do you think about the following instead to reduce code duplication? same with delete logic

```golang
var eniNetworkConfig
if eni-trunking {
  eniNetworkConfig = client.createBranchENINetworkConfig
// maybe also set env variable here
} else {
  eniNetworkConfig = client.createENINetworkConfig
}
// setup ENI
os.setenv(xxxx)
result, err := client.add(runtimeConfig, cfg, eniNetworkConfig)
if err != nil {
  return nil, errors.Wrap(err, "cni setup: invoke %v plugin failed", eniNetworkConfig.Type)
}
seelog.Debugf(xxx)
```

Reply:
This might not work because we are setting different environment variable and setting different error message in the IF blocks. We definitely can save a line when we are setting result, err but I think there is value in keeping it that way in favor of readability.